### ISSUE TITLE
NaN output during ColorizeTrainingStable training

### ISSUE THREAD

**Issue by benswinney** [jantic/DeOldify#121]
I've been dealing with this issue for the past week and I can't seem to get to the bottom of it.

I'm running the ColorizeTrainingStable notebook and training against ImageNet dataset.

Randomly during training I get a NaN output and the training fails.

It can sometimes complete the 64px training, and then NaN on the 128px. I've never been able to pass the 128px training section.

I've tried the following:

Updated to latest fast.ai code
Updated to latest tensorboardX code
Running PyTorch 1.0
Reduced the ImageNet training set to 50K
Reduced it to 39K
Reduced the learning rate
Ran it against multiple GPUs (16Gb P100 / 16Gb V100 / 32Gb V100)
Adjusted batch sizes, with failures even being seen at bs=1
Adjust num_workers=8 / 4 / 2 / 1

All seem to produce the same results.

I should mention this has been ran on a PPC64LE environment, but I don't believe this is the issue.

Can you suggest anything else I could try?

I do get these errors when running the training code:

`/notebooks/datasets/DeOldify/fasterai/unet.py:124: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if ssh != up_out.shape[-2:]:`

`/opt/anaconda3/lib/python3.6/site-packages/torch/onnx/utils.py:501: UserWarning: ONNX export failed on ATen operator reshape because torch.onnx.symbolic.reshape does not exist
  .format(op_name, op_name))
/opt/anaconda3/lib/python3.6/site-packages/torch/onnx/utils.py:501: UserWarning: ONNX export failed on ATen operator mv because torch.onnx.symbolic.mv does not exist
  .format(op_name, op_name))
/opt/anaconda3/lib/python3.6/site-packages/torch/onnx/utils.py:501: UserWarning: ONNX export failed on ATen operator dot because torch.onnx.symbolic.dot does not exist
  .format(op_name, op_name))
/opt/anaconda3/lib/python3.6/site-packages/torch/onnx/symbolic.py:131: UserWarning: ONNX export failed on dim because ONNX and PyTorch use different strategies to split the input. not supported
  warnings.warn("ONNX export failed on " + op + " because " + msg + " not supported")`

Output from last training run:

`

epoch | train_loss | valid_loss | time
-- | -- | -- | --
0 | nan | 3.158493 | 03:54

`

**Comment by jantic** [comment]
Sorry to just get to this now.  The first thing I'd suggest is use the conda install instead of trying to install the latest versions of fastai, tensorboardx, etc.  The conda install specifies specific versions and for good reason:  It's not tested to work with newer packages.  I know for a fact that things keep getting changed in fastai that have the potential to cause issues from time to time (they have for me in the past).  

The Onyx error you're getting there is because of a change in fastai's Tensorboard callback plugin that was added to generate a of the model.  It's annoying but not going to cause actual breaking issues.  I just commented out that stuff myself in the fastai core code.

Finally, have you made any other changes?  Like the loss function, for example?

**Comment by benswinney** [comment]
Unfortunately the conda install can’t be used as I’m running this on a IBM Power9 server with 4 x V100 32GPU’s (ppc64le) and the conda packages have been built for x86 :(

The end result is having to build packages like spaCy etc manually to satisfy the requirements.

It’s bizarre in the fact it will sometimes train the 64px part and then fail at the 128px and then other times fail at the 64px portion.

I will continue digging into things on my end, but if you had any further suggestions that would be great.
